utils/_context/containers.py

```diff
@@ -132,7 +136,39 @@ def stop_previous_container(self):
             logger.debug(f"Kill old container {self.container_name}")
             old_container.remove(force=True)
 
-    def start(self) -> Container:
+    def start(self) -> Thread:
+        return self._start_async([])
+
+    def _start_async(self, seen: list):
+        """ Start the container and its dependencies in a thread with circular dependency detection """
+        if self in seen:
+            dependencies = " -> ".join([s.name for s in seen])
+            pytest.exit(f"Circular dependency detected between containers: {dependencies}", 1)
+
+        seen.append(self)
+
+        with self._starting_lock:
```

Comment:
Small comment won't hurt anybody 

```suggestion
        with self._starting_lock:
            # This lock will prevent any race condition, as this function can be called on the same container in 
            # different threads.
```